File: bench/benches/sum.js
// build the list 1..100000 then sum it -- allocation + traversal.
const { bench } = require("../lib/bench");
const data = Array.from({ length: 100000 }, (_, i) => i + 1);
bench("sum", () => data.reduce((a, b) => a + b, 0));

File: bench/benches/sum.lua
-- build the table 1..100000 then sum it -- allocation + traversal.
package.path = (arg[0]:match("(.*/)") or "./") .. "../lib/?.lua;" .. package.path
local bench = require("bench")
local data = {}
for i = 1, 100000 do data[i] = i end
bench("sum", function()
  local s = 0
  for i = 1, #data do s = s + data[i] end
  return s
end)
